socorro/cron/jobs/monitoring.py

```diff
@@ -0,0 +1,171 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import json
+import os
+from collections import namedtuple
+from os.path import dirname
+from subprocess import PIPE, Popen
+
+from configman import Namespace
+from crontabber.base import BaseCronApp
+
+from socorro.lib import raven_client
+
+
+REPO_ROOT = dirname(dirname(dirname(dirname(__file__))))
+
+
+VulnerabilityBase = namedtuple('Vulnerability', (
+    'type',
+    'dependency',
+    'installed_version',
+    'affected_versions',
+    'description',
+))
+
+
+class Vulnerability(VulnerabilityBase):
+    @property
+    def key(self):
+        return '[%s] %s' % (self.type, self.dependency)
+
+    @property
+    def summary(self):
+        return 'Installed: %s; Affected: %s; %s' % (
+            self.installed_version,
+            self.affected_versions,
+            self.description,
+        )
+
+
+class DependencySecurityCheckFailed(Exception):
+    """Thrown when a security check cannot complete, such as network
+    issues.
+
+    """
+
+
+class DependencySecurityCheckCronApp(BaseCronApp):
+    """Configuration values used by this app:
+
+    crontabber.class-DependencySecurityCheckCronApp.node_modules
+        Path to the node_modules directory where the webapp's npm
+        dependencies have been installed.
+    secrets.sentry.dsn
+        If specified, vulnerabilities will be reported to Sentry instead
+        of logged to the console.
+
+    """
+    app_name = 'dependency-security-check'
+    app_description = (
+        'Runs third-party tools that check for known security vulnerabilites in Socorro\'s '
+        'dependencies.'
+    )
+    app_version = '0.1'
+
+    required_config = Namespace()
+    required_config.add_option(
+        'node_modules',
+        doc=(
+            'Path to node_modules directory where the webapp\'s npm dependencies have been '
+            'installed.'
+        ),
+    )
+
+    def run(self):
```

Comment:
What happens if `node_modules` isn't configured? I have vague memories of bumping into that and having to write an explicit check to make sure it has a value.